esmvalcore/preprocessor/_multimodel.py

```diff
@@ -295,74 +291,55 @@ def _assemble_data(cubes, statistic, span='overlap'):
     return stats_cube
 
 
-def multi_model_statistics(products, span, statistics, output_products=None):
-    """Compute multi-model statistics.
+def multicube_statistics(cubes, statistics, span):
+    """Compute statistics across input cubes.
 
-    Multimodel statistics computed along the time axis. Can be
-    computed across a common overlap in time (set span: overlap)
-    or across the full length in time of each model (set span: full).
-    Restrictive computation is also available by excluding any set of
-    models that the user will not want to include in the statistics
-    (set exclude: [excluded models list]).
+    This function deals with non-homogeneous cubes by taking the time union
+    (span='full') or intersection (span='overlap'), and extending or subsetting
+    the cubes as necessary. Apart from the time coordinate, cubes must have
+    consistent shapes.
 
-    Restrictions needed by the input data:
-    - model datasets must have consistent shapes,
-    - higher dimensional data is not supported (ie dims higher than four:
-    time, vertical axis, two horizontal axes).
+    This function operates directly on numpy (masked) arrays and rebuilds the
+    resulting cubes from scratch. Therefore, it is not suitable for lazy
+    evaluation. This function is restricted to maximum four-dimensional data:
+    time, vertical axis, two horizontal axes.
```

Comment:
I think the old docstring was explaining the functionality very well using terms that a climate scientist would understand better than this changed one, I don't get why make things very technical when there is no need to, remember this is read by not so technical people via the API documentation

Reply:
Good point about the intended audience. 

This is in anticipation of #950, where I implement an altnerative function that does exactly the same thing in a different way. But I'll rephrase it for this PR.

Reply:
cheers :beer:

Reply:
The old docstring is back in the public API function below.